dali/operators/image/crop/bbox_crop.cc

```diff
@@ -617,11 +683,20 @@ class RandomBBoxCropImpl : public OpImplBase<CPUBackend> {
 
   bool ValidOverlap(const Box<ndim, float> &crop,
                     span<const Box<ndim, float>> boxes,
-                    float threshold) {
-    return std::all_of(boxes.begin(), boxes.end(),
-      [&crop, threshold](const Box<ndim, float> &box) {
-        return intersection_over_union(crop, box) >= threshold;
-      });
+                    float threshold, ThresholdType threshold_type) {
+    std::function<bool(const Box<ndim, float> &box)> f;
+    if (threshold_type_ == ThresholdType::Overlap) {
+      f = [&crop, threshold](const Box<ndim, float> &box) {
```

Comment:
It's cheaper to call all_of in each branch than to use std::function - last time I checked it was a relatively heavy object and had onsiderable call overhead.

Reply:
Because I'd like to have the option to use ``all_of`` and ``any_of``, I will keep the `std::function`